Beta_Integrations/integration-Vertica.yml

```diff
@@ -0,0 +1,178 @@
+    PASSWORD = demisto.params().get('credentials').get('password')
+    DATABASE = demisto.params().get('database')
+    PORT = int(demisto.params().get('port', 5433))
+    SERVER = demisto.params()['url'][:-1] if (demisto.params()['url'] and demisto.params()['url'].endswith('/')) else demisto.params()['url']
+    DB_PARAMS = {
+        'host': SERVER,
+        'port': PORT,
+        'user': USERNAME,
+        'password': PASSWORD,
+        'database': DATABASE,
+        'connection_timeout': 5
+    }
+
+    ''' HELPER FUNCTIONS '''
+
+    def convert_datetime_to_string(v):
+        """
+        Parses datetime object into string
+        """
+        if isinstance(v, datetime.datetime):
+            return v.strftime("%Y-%m-%dT%H:%M:%S")
+        return v
+
+
+    def connect_db():
+        try:
+            connection = vertica_python.connect(**DB_PARAMS)
+        except vertica_python.errors.ConnectionError:
+            return_error('Could not connect to DB, please re-check DB params.')
+        return connection
+
+
+    ''' COMMANDS + QUERY FUNCTIONS '''
+
+    def test_module():
+        """
+        Performs basic query on default system tables
+        """
+        cursor.execute("SELECT * FROM system_tables ORDER BY table_schema, table_name LIMIT 2;")
+        rows = cursor.fetchall()
+        if cursor.rowcount == 0:
+            return_error('No results were returned from the DB.')
+        demisto.results('ok')
+
+
+    def query_command():
+        """
+        Execute a query against the DB
+        """
+        # Init main vars
+        contents = []
+        context = {}
+        title = 'Vertica Query'
+
+        # Get arguments from user
+        query = demisto.args().get('query')
+        limit = int(demisto.args().get('limit', 50))
+        # Query and get raw response (list of ordered dicts)
+        rows = query_request(query)
+
+        # Parse response into context & content entries
+        if rows:
+            if limit:
+                rows = rows[:limit]
+            title = 'Vertica Query Results:'
+
+            for i, row in enumerate(rows):
+                rows[i] = { underscoreToCamelCase(k): convert_datetime_to_string(v) for k, v in row.items() }
+
+            contents = rows
+            context['Vertica.Query(val.Query && val.Query === obj.Query)'] = {
+                'Query': query,
+                'Rows': rows
+            }
+
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'Contents': contents,
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown(title, contents, removeNull=True),
+            'EntryContext': context
+        })
+
+
+    def query_request(query):
+        try:
+            cursor.execute(query)
+        except vertica_python.errors.MissingRelation:
+            return_error('Error while executing query.')
```

Comment:
saw your reply to rony, but are there any details we can add here?
not sure what to do when such an error occurs ?

Reply:
There is some issue with the SDK itself when trying to extract that info, it claims the data is not accessible, I have trying to diagnose that using dir()/getattr() on the MissingRelation object, but it simply didn't work.